libs/editor/editor_notes.cpp

```diff
@@ -155,12 +153,21 @@ void Notes::CreateNote(ms::LatLon const & latLon, std::string const & text)
 
 void Notes::Upload(osm::OsmOAuth const & auth)
 {
+  if (m_isUploadingNow)
+    return;
+  m_isUploadingNow = true;
+
   // Capture self to keep it from destruction until this thread is done.
-  auto const self = shared_from_this();
+  /// @todo It looks redundant because of parent static Editor instance.
+  auto self = shared_from_this();
 
-  auto const doUpload = [self, auth]()
+  GetPlatform().RunTask(Platform::Thread::Network, [self = std::move(self), auth]()
```

Comment:
On both iOS and Android, Notes::Upload is already called in a background thread context. These async calls were likely to "upload notes and edits in parallel".

A more reliable approach is to remove any threading here, make notes uploading single-threaded, and launch it completely independent from edits from Editor::UploadChanges(). Or don't launch in parallel at all to avoid issues with iOS and Android background task stopping, as it is not easy to interrupt a separate thread when iOS or Android requires it.

Reply:
Its a simple and _reliable_ approach for the upcoming release that fixes an obvious problem. You can refactor as you like after.